### DISCUSSION TITLE
override domain question

### DISCUSSION THREAD

**Discussion by anispwyn** [diced/zipline#475] [Questions] [answered]
so i have this domain 
lets call it zip.example.com

in the manage user i override the domain from zip.example.com to files.example.com

but for some reason files.example.com just showing blank pages
how do you usually set this up?

i'm using caddy and here's my caddyfile

```

*.example.com {
        tls {
                dns cloudflare {env.Cloudflare_api_token}
        }
        @zipline host z.example.com
                handle @zipline {
                reverse_proxy 127.0.0.1:3000
        }
}

```

**Comment by diced** [comment, marked as answer]
i dont know how caddy works so I'll try explaining in a way that is general I guess. lets say you have your main domain you send upload requests from: `img.example.com` [POST https://up.example.com/api/upload, Headers: (all the necessary shit + `Override-Domain: file.example.com`)] - this will output `https://file.example.com/u/file.ext` instead of `https://up.example.com/u/file.ext` (the default).

to make this work you need to make sure that up.example.com points to the same adress as file.example.com, whether it's through your registrars dns or caddy/nginx/apache (or anything else). 

for example, since im familiar with nginx you would do the following:
```nginx
server {
  listen 80 default_server;
  client_max_body_size 100M;
  server_name up.example.com file.example.com;
  
  location / {
    proxy_pass http://localhost:3000;
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto $scheme;
  }
}
```

then in my dns settings in cloudflare or my registrar i would put
`up.example.com  A  123.123.123.123`
`file.example.com A  123.123.123.123`

it's important that they point to the same ip, as previously stated!

**Reply by anispwyn** [reply]
From the look of the nginx config

I assume i just have to reverse proxy zipline into two subdomains right? (Also nginx config works fine hmbecause caddy have nginx adapt (convert nginx to caddy json))

**Reply by diced** [reply]
>I assume i just have to reverse proxy zipline into two subdomains right?

yes
